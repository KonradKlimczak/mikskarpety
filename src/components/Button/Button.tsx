import classnames from 'classnames';
import React, { ButtonHTMLAttributes, FunctionComponent } from 'react';
import { Size, getSizeClassName } from '../../utils/Size';

interface IButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  primary?: boolean;
  round?: boolean;
  size?: Size;
}

const Button: FunctionComponent<IButtonProps> = (props) => {
  const { primary, round, size, ...buttonProps } = props;
  return (
    <button
      {...buttonProps}
      className={classnames('button', {
        ['button-primary']: primary,
        ['button-round']: round,
      }, getSizeClassName(size))}
    >
      {props.children}
    </button>
  );
};

export default Button;
