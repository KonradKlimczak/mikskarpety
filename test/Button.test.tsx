import {} from 'jest';
import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../src/components/Button/Button';

it('Button renders correctly', () => {
  const tree = renderer.create(<Button>Test</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Button add classes', () => {
  const primary = renderer.create(<Button primary>Test</Button>).toJSON();
  expect(primary).toMatchSnapshot();
  const round = renderer.create(<Button round>Test</Button>).toJSON();
  expect(round).toMatchSnapshot();
});
