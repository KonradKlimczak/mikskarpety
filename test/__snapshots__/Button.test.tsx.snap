// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button add classes 1`] = `
<button
  className="button button-primary size--medium"
>
  Test
</button>
`;

exports[`Button add classes 2`] = `
<button
  className="button button-round size--medium"
  round={true}
>
  Test
</button>
`;

exports[`Button renders correctly 1`] = `
<button
  className="button size--medium"
>
  Test
</button>
`;
